### DISCUSSION TITLE
Partial watch mode

### DISCUSSION THREAD

**Discussion by wujekbogdan** [vercel/turborepo#8318] [Ideas]
### Goals

Being able to use a bundler-based watcher for a certain monorepo package and turbo watch for all its monorepo dependencies.

### Background

I have a monorepo with several apps that rely on workspace dependencies. Thanks to the newly introduced watch mode, I can run a dev mode for my app that recompiles whenever its monorepo dependencies change, which is great.

The thing is, for the app I'm developing, I would rather use the built-in watcher (because of hot reloading and other goodies) rather than the turbo watch features. For everything else, I would like to rely on `turbo watch`.

E.g., when running:

```
turbo watch dev --filter=@my-namespace/my-app
```

that has a dev task defined as:

```
"dev": "tsup --watch --onSuccess 'node dist/index.js'`
```

I would like the `@my-namespace/my-app app` to be excluded from turbo watch because it already has a bundler that handles watching. However, I would still like all its monorepo dependencies to be re-built on change thanks to `turbo watch`.

### Proposal

I'm not sure what the API for this feature would look like. I think we would need an additional CLI flag or maybe a watch blacklist in the `turbo.json` file.

**Comment by ThallesP** [comment]
I really would need this feature! In my case I would like that my Prisma package auto builds and compiles when any changes happens inside the package.

**Comment by anthonyshew** [comment]
Persistent tasks are already ignored by `watch`: https://turbo.build/repo/docs/reference/watch#using-turbo-watch-with-persistent-tasks

Are the tasks not marked as persistent?
